### PR TITLE
I-#118: Added Localstack pro auth token

### PR DESCRIPTION
**Pull request by mahinuzzaman**
Feat: #118 

Add LocalStack Pro access in the GitHub CI workflow.

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [ ] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?

Provide a brief overview of the changes introduced in this pull request.

## Related Issues/PRs (optional)

Link to the relevant issue(s) or provide a clear description of the problem or feature being addressed.

## How to test

- [ ] Needs regression testing (add QA needed label)

## Checklist

- [x] I have tested these changes locally and they function as expected.
- [ ] I have added or updated necessary documentation to reflect the changes.
- [ ] I have followed the project's coding style and guidelines.
- [ ] I have added appropriate test coverage to ensure the changes are functioning correctly.
- [ ] I have checked for any potential conflicts with other branches or pull requests.
- [ ] I have updated necessary OpenAPI specs

## Reviewer Request (optional)

If you have any specific requests or suggestions for reviewers, mention them here.

## Screenshots (optional)

If applicable, provide any additional screenshots or GIFs that showcase the changes made in this pull request.

## Demo Link (optional)

If applicable, provide a link to a live demo or deployment where the changes can be reviewed.

## Known Issues (optional)

If there are any known issues or limitations with the changes made in this pull request, describe them here.

## Additional Notes (optional)

Add any additional notes or comments you think are relevant to this pull request.

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -104,13 +126,18 @@ build-runner-image:
...
+	           $(DOCKER_RUN_MOUNT_OPTIONS) $(DOCKER_ENV) $(RUNNER_IMAGE_NAME) \
+	           make $* EXTRA_ARGS=$(EXTRA_ARGS)
 
 # Dev commands
 start-dev: build-runner-image localstack-start run-command-install-node-packages run-dev
 
-run-dev: run-command-build-node-all run-command-package-all run-command-tf-init run-command-tf-plan-apply run-command-tf-apply
+run-dev: run-command-build-node-all run-command-package-all run-command-tf-init \
+         run-command-tf-plan-apply run-command-tf-apply
+
```
Comment (KhirulIslam) [inline Makefile:138]: please remove extra line. One line gap, should be enough

File: Makefile [hunk 2]
```diff
@@ -22,13 +44,13 @@ bundle-openapi:
 	redocly bundle openapi/versions/v1.json -o docs/openapi/v1.json
 
 
-# Deployment
+# Terraform base command
```
Comment (KhirulIslam) [inline Makefile:47]: please add bit more context on why we are doing this.

File: Makefile [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+            -e TF_BACKEND_BUCKET_NAME \
+            -e TF_BACKEND_BUCKET_REGION \
+            -e TF_BACKEND_BUCKET_KEY $(TF_VARS)
+
+# Terraform Backend Configuration
+TF_BACKEND_CONFIG=--backend-config="bucket=$(TF_BACKEND_BUCKET_NAME)" \
+                  --backend-config="key=$(TF_BACKEND_BUCKET_KEY)" \
+                  --backend-config="region=$(TF_BACKEND_BUCKET_REGION)"
+
+# Checkov Skip Rules
```
Comment (KhirulIslam) [inline Makefile:30]: please add why we are skipping these rurles

**Review by KhirulIslam — COMMENTED** [review]
Let's check for logs if we are echoing LOCALSTACK_AUTH_TOKEN anywhere in CI pipeline ..

File: Makefile (hunk 3 above)
Comment (mahinuzzaman) [reply Makefile:30]: updated in b6b119a6dd73086ee41929ddbaedf2483cca9b9c

File: Makefile (hunk 1 above)
Comment (mahinuzzaman) [reply Makefile:146]: updated in b6b119a6dd73086ee41929ddbaedf2483cca9b9c

File: Makefile (hunk 2 above)
Comment (mahinuzzaman) [reply Makefile]: updated in b6b119a6dd73086ee41929ddbaedf2483cca9b9c

**Comment by mahinuzzaman** [comment]
> Let's check for logs if we are echoing LOCALSTACK_AUTH_TOKEN anywhere in CI pipeline ..

It looks safe in logs
image

**Review by KhirulIslam — APPROVED** [review]

**Pull request merged**